src/fetchStats.js

```diff
@@ -36,9 +36,45 @@ async function fetchStats(username) {
           }
         }
       }
-    `,
-    variables: { login: username },
-  });
+      `,
+      variables: { login: username },
+    },
+    {
+      // set the token
+      Authorization: `bearer ${token}`,
+    }
+  );
+};
+
+async function retryer(username, RETRIES) {
+  try {
+    console.log(`Trying PAT_${RETRIES + 1}`);
+
+    // try to fetch with the first token since RETRIES is 0 index i'm adding +1
+    let response = await fetcher(username, process.env[`PAT_${RETRIES + 1}`]);
+
+    // if rate limit is hit increase the RETRIES and recursively call the retryer
+    // with username, and current RETRIES
+    if (
+      response.data.errors &&
+      response.data.errors[0].type === "RATE_LIMITED"
+    ) {
+      console.log(`PAT_${RETRIES} Failed`);
+      RETRIES++;
+      // directly return from the function
+      return await retryer(username, RETRIES);
+    }
+
+    // finally return the response
+    return response;
+  } catch (err) {
+    console.log(err);
+  }
+}
```

Comment:
main meat!

Reply:
Looks good to me, 

Maybe a check to see if all tokens have hit the rate limit would be nice, though. Because it will keep looping over all tokens and possibly cause some performance issues.

Reply:
Yup that would work but dont need to do that because i can just fallback to another token if one fails, this way i dont have to recheck if token is valid

Reply:
Yup, but what if all tokens hit the limit at the same time (difficult at the moment)? Is this something that could happen?

I'm just concerned that, if all tokens hit the limit, the recursive function will run forever, well, until a token has the rate limit reset. And might give some performance issues.

Can this recursive function run forever in that scenario? If it does not, there is no problem.

Reply:
Well its very unlikely that all tokens will expire at once, because it takes time to expire one token and all tokens will be revalidated after 1hours and i will have a condition to prevent infinite loops

```js
if (RETRIES > 7) {
    throw new Error("Maximum retries exceeded");
  }

```

Reply:
Ohh okay, no issue then. That will do the job nicely! I guessed it would be very unlikely but just in case 👍